### PR TITLE
[tests] Fix device tests to run on arm64

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

Right now the device tests assume an x64 host. This PR will enable running on arm64 hosts.

### CODE REVIEW

**Review by rmarinho — CHANGES_REQUESTED** [review]
It's failing to provision the android images on windows.

https://dev.azure.com/xamarin/public/_build/results?buildId=91728&view=logs&j=22083a8f-f531-5d1a-9d1c-44debe18920a&t=aef5f1be-4bed-5de7-be44-f42d3ac5bea4

**Comment by rmarinho** [comment]
It works!

**Review by rmarinho — APPROVED** [review]

File: eng/devices/android.cake [hunk 1]
```diff
@@ -82,25 +82,33 @@ Setup(context =>
...
 				DEVICE_ARCH = "x86";
 			else
 				DEVICE_ARCH = "armeabi-v7a";
 		} else if (parts[2] == "64") {
-			if (emulator)
+			if (System.Runtime.InteropServices.RuntimeInformation.OSArchitecture == System.Runtime.InteropServices.Architecture.Arm64)
+				DEVICE_ARCH = "arm64-v8a";
+			else if (emulator)
 				DEVICE_ARCH = "x86_64";
 			else
```
Comment (sbanni) [inline eng/devices/android.cake:103]: Can we just have the one if (emulator) and then else to arm64?

**Review by sbanni — APPROVED** [review]

**Pull request merged**